src/contracts/entrypoints.rs

```diff
@@ -12,22 +12,63 @@ use ink::primitives::AccountId;
 
 #[ink::trait_definition]
 pub trait Invariant {
+    /// Retrieves the protocol fee represented as a percentage.
     #[ink(message)]
     fn get_protocol_fee(&self) -> Percentage;
 
+    /// Allows an authorized user to withdraw collected fees.
+    ///
+    /// # Parameters
+    /// - `pool_key`: A unique key that identifies the specified pool.
+    ///
+    /// # Errors
+    /// - Reverts the call when the caller is an unauthorized receiver.
     #[ink(message)]
     fn withdraw_protocol_fee(&mut self, pool_key: PoolKey) -> Result<(), InvariantError>;
 
+    /// Allows an authorized user to adjust the protocol fee.
```

Comment:
the only user that is allowed to change protocol fee is an admin so to be more specific "authorized user" should be changed to "admin"